torch_geometric/llm/models/llm.py

```diff
@@ -135,22 +181,158 @@ def __init__(
                 "was made by a rough hueristic that assumes your GPU set up "
                 "does not have enough GPU RAM. This is done to avoid GPU OOM "
                 "errors. If you think this is a mistake, please initialize "
                 "your LLM with the n_gpus param to dictate how many gpus to "
-                "use for the LLM.", stacklevel=2)
-            self.device = torch.device('cpu')
+                "use for the LLM.",
+                stacklevel=2,
+            )
+            self.device = torch.device("cpu")
             self.autocast_context = nullcontext()
         else:
             self.device = self.llm.device
             if dtype == torch.float32:
                 self.autocast_context = nullcontext()
             else:
-                self.autocast_context = torch.amp.autocast('cuda', dtype=dtype)
+                self.autocast_context = torch.amp.autocast("cuda", dtype=dtype)
+
+        # ---- Uncertainty config / backends (opt-in) ----------------------
+        self.uncertainty_estim = bool(uncertainty_estim)
+        self._uncfg: Dict[str, Any] = {
+            "h_star": 0.05,
+            "isr_threshold": 1.0,
+            "m": 6,
+            "n_samples": 3,
+            "B_clip": 12.0,
+            "clip_mode": "one-sided",
+            "skeleton_policy": "auto",
+            "q_floor": None,
+            "temperature": 0.5,
+            "max_tokens_decision": 8,
+            "backend": "hf",  # "hf" | "ollama" | "anthropic"
+            "mask_refusals_in_loss": True,
+        }
+        if uncertainty_cfg:
+            self._uncfg.update(uncertainty_cfg)
+        self._dec_backend_kwargs = decision_backend_kwargs or {}
+        self._last_uncertainty: Optional[List[_Uncertainty]] = None
+
+        # Try to import your planner/backends; fall back to a local adapter:
+        self._planner = None
+        self._backend = None
+        if self.uncertainty_estim:
+            try:
+                from hallbayes import OpenAIItem, OpenAIPlanner
+
+                self._OpenAIPlanner = OpenAIPlanner
+                self._OpenAIItem = OpenAIItem
+                try:
+                    from hallbayes import (
+                        AnthropicBackend,
+                        HuggingFaceBackend,
+                        OllamaBackend,
+                    )
+
+                    self._HFBackend = HuggingFaceBackend
+                    self._OllamaBackend = OllamaBackend
+                    self._AnthropicBackend = AnthropicBackend
+                except Exception:
+                    self._HFBackend = self._OllamaBackend = (
+                        self._AnthropicBackend) = None
+            except Exception:
+                self.uncertainty_estim = False
+                warnings.warn(
+                    "Uncertainty requested but `hallucination_toolkit` not "
+                    "importable. Proceeding without uncertainty.",
```

Comment:
can you add a copypastable string for the pip install users should use to get it set up